lib/AuthStateManager.ts

```diff
@@ -44,15 +46,31 @@ export class AuthStateManager {
     this._pending = { ...DEFAULT_PENDING };
     this._authState = { ...DEFAULT_AUTH_STATE };
     this._logOptions = {};
+    this._lastEventTimestamp = 0;
 
     // Listen on tokenManager events to start updateState process
-    // "added" event is emitted in both add and renew process, just listen on "added" event to update auth state
-    sdk.tokenManager.on('added', (key, token) => {
-      this._setLogOptions({ event: 'added', key, token });
+    // "added" event is emitted in both add and renew process
+    // Only listen on "added" event to update auth state
+    const shouldUpdateAuthState = (timestamp: number): boolean => {
```

Comment:
Do we actually need this logic? If updateAuthState is called repeatedly or unnecessarily it should not cause any negative side-effects. If it produces an identical state to the current state it will not emit a change, so will it actually matter if an "older" event was the reason it is being called?

Reply:
> so will it actually matter if an "older" event was the reason it is being called

Yes, the logic here is to handle the 1s delay hack for IE.

Reply:
Yeah its a hack to call the events later, but does that actually cause a problem? And if it doesn't cause a problem why do we need extra logic to suppress the calls?

Reply:
Other than the cross tabs update, each tab is still running autoRenew process by itself. In this case, the delayed event will override the value from the tab's own autoRenew result if it comes later.

Reply:
I don't understand "override the value". Isn't it safe to call `updateAuthState` at any time? It will read tokens from storage and, if these are different than what is currently in the authState, then it will emit a change event. If there is no change in the state, then nothing is emitted and it is essentially a no-op.  If 1 second after an auto renew, the values in storage have changed, then we want this change event, right?